### PR TITLE
fix(Core/Spells): Passive auras with SPELL_ATTR0_ONLY_INDOORS/SPELL_A…

### LINKED ISSUE #9209: Druid Dragonhide Set Speed Bonus Not Updating Correctly When Going Outside
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [X] 50-59

### Current Behaviour
Original report: https://github.com/chromiecraft/chromiecraft/issues/2340

The PvP Druid 4set bonus increases your movement speed in bear, cat and travel form by 15%. And it only works outdoors. It works just fine but, when you enter the building in bear/cat form the bonus disappear as it should according to the describe of that set bonus, but when you leave the building you need to reshift bear/cat form (not talking about travel form because you can use that form only outdoors) to gain the bonus again.
4set bonus

Tester: Confirmed. I used the following macro to track current movement speed:
```/script print(string.format("Current speed: %d%%", GetUnitSpeed("player") / 7 * 100))```

Cat test - starts outside at 149% speed. Going inside correctly drops it to 129%, and going back outside shows it is still 129%. Only shifting corrects the speed back to 149%:

https://user-images.githubusercontent.com/81782124/141972107-38ce2efa-179d-49a0-bd05-f17bc90f65b8.mp4

Bear test - starts at 115%, correctly drops to 100% inside, remains there even after going back outside, and only returns to correct value after shifting:

https://user-images.githubusercontent.com/81782124/141972537-703f76eb-92d3-4529-b4e6-65f9e18d8c55.mp4

### Expected Blizzlike Behaviour

You should gain the bonus movement speed when you leave the building automatically, without needing to reshift your bear/cat form.

### Source

No direct source found, but I think basic consistency says this should not be occurring.

### Steps to reproduce the problem

Get the set, which is actually bit of a pain due to rep requirements needed to wear it:

.additemset 742
.modify reputation 947 9000
.modify reputation 935 9000
.modify reputation 942 9000
.modify reputation 1011 9000
.modify reputation 989 9000

Add speed checker macro somewhere: ```/script print(string.format("Current speed: %d%%", GetUnitSpeed("player") / 7 * 100))```
Run into the building in bear/cat form with 4set PvP bonus
Leave the building
Not getting the set bonus, you need to reshift bear/cat form instead of getting bonus automatically

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/03c84db04dd7bce2a8aa5e527aeec2640c6fa7eb

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

### PR DESCRIPTION
**Pull request by UltraNix**
…TTR0_ONLY_OUTDOORS should not be removed when entering/leaving buildings.

Fixes #9209

## Issues Addressed:

- Closes #9209
- Closes https://github.com/chromiecraft/chromiecraft/issues/2340

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

.additemset 742
.modify reputation 947 9000
.modify reputation 935 9000
.modify reputation 942 9000
.modify reputation 1011 9000
.modify reputation 989 9000

Add speed checker macro somewhere: /script print(string.format("Current speed: %d%%", GetUnitSpeed("player") / 7 * 100))
Run into the building in bear/cat form with 4set PvP bonus
Leave the building
Not getting the set bonus, you need to reshift bear/cat form instead of getting bonus automatically

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by UltraNix** [comment]
@azerothcore/testers Please, test it.

**Comment by temperrr** [comment]
tested, works :)

**Review by Nyeriah — APPROVED** [review]

**Comment by Nyeriah** [comment]
Codestyle check script:

  Checking RegEx: '\t'
  Checking RegEx: 'LOG_.+GetCounter'
  Checking RegEx: '[[:blank:]]$'
src/server/game/Entities/Player/Player.h:2963:    

Remove whitespace at the end of the lines above

File: src/server/game/Entities/Player/Player.h [hunk 1]
```diff
@@ -2959,6 +2959,8 @@ class Player : public Unit, public GridObject<Player>
 
     Optional<float> _farSightDistance = { };
 
+    bool _wasOutdoor;
+
```
Comment (acidmanifesto) [inline src/server/game/Entities/Player/Player.h:2963]: remove the blank space on line 2963.
image

**Review by acidmanifesto — CHANGES_REQUESTED** [review]

File: src/server/game/Entities/Player/Player.h (hunk 1 above)
Comment (Nyeriah) [inline src/server/game/Entities/Player/Player.h:2963]:
```suggestion
    bool _wasOutdoor;

```

**Pull request merged**